mirror_builder/overrides/maturin.py

```diff
@@ -0,0 +1,19 @@
+import logging
+
+# import shutil
+
+logger = logging.getLogger(__name__)
+
+
+# FIXME: Using system packages for some dependencies like maturin may
+# not work if the system package is not available for the version of
+# Python for which we're building the wheel.
```

Comment:
I've added some logic to try to use system packages for some dependencies, but it does not work because the system package only has files for one version of Python but we may need to build wheels for multiple versions.

Reply:
I don't know anything about maturin or building Rust crates, but it does appear that it has support for multiple Python version:

```
$ sudo dnf install python3.8 python3.9 python3.10 python3.11
$ maturin list-python
🐍 5 python interpreter found:
 - CPython 3.8 at /usr/bin/python3.8
 - CPython 3.9 at /usr/bin/python3.9
 - CPython 3.10 at /usr/bin/python3.10
 - CPython 3.11 at /usr/bin/python3.11
 - CPython 3.12 at /usr/bin/python3.12
```

So that's promising?

But I think we need to discuss our multiple-python-versions plan a bit more. Just added this to [our doc](https://hackmd.io/uae_7T7rRqyWphoCV5dB4w):

> The prototype builds wheels for multiple versions of Python, but what will make sense in practice? Should an index for a given Fedora version be expected to work with the default version of Python in that version of Python? Or all versions of Python available in that version of Fedora? If we did decide to support multiple Python versions for a given Fedora version, should we have a separate set of wheels for each version?